Azure Purview Policy Store Self-service Policies Spec

saranya-azuredev opened the pull request:
# Data Plane API - Pull Request
<!-- 🚨🚨🚨 Replace this line with a summary and reason for these changes to your data plane API 🚨🚨🚨 -->

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*engagement record*](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/271/Azure-REST-API-Stewardship?anchor=rest-api-stewardship-process).

* Link to engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous Open API document (swagger) if applicable, and the root paths that have been updated. </sup>
* Design Document:
* Previous Open API Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azapi) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azapi).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

saranya-azuredev committed:
Specs for Self Service Policies

saranya-azuredev committed:
Fix Prettier

saranya-azuredev committed:
readme file changes

saranya-azuredev committed:
Resolving lint issues

saranya-azuredev committed:
Modify next link value

saranya-azuredev commented:
@mikekistler kindly review the PR

mikekistler commented on specification/purview/data-plane/Azure.Analytics.Purview.SelfServicePolicies/preview/2022-12-01-preview/purviewSelfServicePolicy.json:
```diff
@@ -0,0 +1,348 @@
+      ],
+      "type": "object",
+      "properties": {
+        "workflowRunId": {
```
You should not call this property `workflowRunId` or describe it as the id of a workflow. This is the id of the selfServicePolicy so it should be called something like `policyId`. Your service might derive it from the workflowRunId but this is an internal detail that should not leak out into the API contract.

saranya-azuredev replied:
Have modified the field to policyId instead of workflowRunId.

mikekistler commented on specification/purview/data-plane/Azure.Analytics.Purview.SelfServicePolicies/preview/2022-12-01-preview/purviewSelfServicePolicy.json:
```diff
@@ -0,0 +1,348 @@
+          "type": "string",
+          "description": "The requestor of the self-service policy"
+        },
+        "expiryDate": {
```
It is good practice to have a naming convention for date-time properties (we are working an update to the Azure REST API guidelines for this). This file has 3 date-time properties and the other two end with "At", so perhaps this property could be named "expiresAt".

mikekistler commented on specification/purview/data-plane/Azure.Analytics.Purview.SelfServicePolicies/preview/2022-12-01-preview/purviewSelfServicePolicy.json:
```diff
@@ -0,0 +1,348 @@
+    "SystemData": {
+      "type": "object",
+      "description": "The system data",
+      "readOnly": true,
```
`readOnly: true` has no significance for schemas that are only used in responses, so you should remove this.

mikekistler commented on specification/purview/data-plane/Azure.Analytics.Purview.SelfServicePolicies/preview/2022-12-01-preview/purviewSelfServicePolicy.json:
```diff
@@ -0,0 +1,348 @@
+      "required": true,
+      "type": "string",
+      "x-ms-client-name": "ApiVersion",
+      "x-ms-parameter-location": "method"
```
Why `x-ms-parameter-location: method` for api-version? This typically goes on the client.

saranya-azuredev replied:
Modified x-ms-parameter-location to client instead of method.

mikekistler reviewed (commented):
Style-wise this looks very nice! I noted a few minor points that should be easy to fix.

saranya-azuredev committed:
Address API review comments

  1) Changed workflowRunId parameter to policyId
  2) Changed expiryDate to expiresAt
  3) ApiVerion changed from method to get from client.

mikekistler commented:
/azp run

saranya-azuredev committed:
Address API review comments

1) Make "principal" and "principalGroup" an array rather than a single value to be consistent with Devops policies.
2) Remove System.string from policyId to make it language agnostic.

mikekistler reviewed (approved):
This looks good for preview. 👍
